File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
          long temp=x,sum=0;
        while(x>0){
            int dig=x%10;
            sum=sum*10+dig;
            x=x/10;
        }
        if(sum==temp)
            return true;
        else
        return false; 
    }
};

   

File: 9-palindrome-number/palindrome-number.java
class Solution{
public boolean isPalindrome(int x){ 
        int original=x;
        int rev=0;
        if(x<0)return false;
        while(x>0){
            int digit=x%10;
            rev=rev*10+digit;
            x=x/10;
        }

        return original == rev;
    }
};
